db/versions/0009.yml

```diff
@@ -212,3 +212,92 @@ methods:
           end;
         end if;
       end
+  purge_requests:
+    description: |-
+      List the caches for this `provisionerId`/`workerType` that should be
+      purged if they are from before the time given in the response.
+      This is intended to be used by workers to determine which caches to purge.
+    mode: read
+    serviceName: purge_cache
+    args: provisioner_id_in text, worker_type_in text
+    returns: table (provisioner_id text, worker_type text, cache_name text, before timestamptz)
+    body: |-
+      begin
+        return query
+        select cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name, cache_purges.before from cache_purges
+        where cache_purges.provisioner_id = purge_requests.provisioner_id_in and cache_purges.worker_type = purge_requests.worker_type_in;
+      end
+  all_purge_requests:
+    description: |-
+      View all active purge requests.
+      This is useful mostly for administors to view
+      the set of open purge requests. It should not
+      be used by workers. They should use the purgeRequests
+      endpoint that is specific to their workerType and
+      provisionerId.
+    mode: read
+    serviceName: purge_cache
+    args: page_size_in integer, page_offset_in integer
+    returns: table (provisioner_id text, worker_type text, cache_name text, before timestamptz)
+    body: |-
+      declare
+      sql text := 'select cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name, cache_purges.before from cache_purges';
+      begin
+        return query select cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name, cache_purges.before
+        from cache_purges
+        order by cache_purges.provisioner_id, cache_purges.worker_type, cache_purges.cache_name
+        limit case
+          when (page_size_in is not null and page_size_in > 0) then page_size_in
+          else null
+        end
+        offset case
+          when (page_offset_in is not null and page_offset_in > 0) then page_offset_in
+          else 0
+        end;
```

Comment:
Since you're defining an earlier version than my #2748 is, do you want to factor these out into a utility function (defined in the migration script) like you suggested today?  Then I'll modify #2748 to use those utility functions.